course-outline.md

```diff
@@ -0,0 +1,105 @@
+
+# Suggested Learning Objectives / Outline of Each Section
+
+The overall idea of this course can be articulated as introducing students to enough tooling knowledge that they are comfortable creating and maintaining an online coding/technical resource, such as a JupyterBook. This will cover the skills they need to work with any type of literate coding document under version control:
+
+- Use the shell to move around on their computer
+- Use git and github to version control their project
+- Have a good project organization
+- Install jupytet-book in an isolated virutal env
+- Create reproducible notebook documents and version control them
+- Build and upload a Jupyter Book
+
+## 1. Intro
+
+1. Introduction to the toolbox, give an overview of what each tool does and where it fits in the toolchain
+2. Intro to VS Code
+    - I think this makes sense because it lets us use vscode for markdown documents, merge conflicts, and graphical git views in the following chapters. It can also be used to open files from terminal via `code <filename>`, which jupyterlab can't.
+        - I am hesitant to fully teach the graphical git interface from vscode and jupyterlab, but we can discuss it.
+3. Create minimal reproducible code examples when asking for help.
+    - [Lec8 521 first half](https://pages.github.ubc.ca/mds-2021-22/DSCI_521_platforms-dsci_students/materials/lectures/8-asking-effectively-and-virtual-environments.html)
+
+- Assignment 1 will be [Installing things](https://ubc-mds.github.io/resources_pages/installation_instructions/)
+    - We could either give this as part of lab0 only, or show how to do some of it during lecture as well.
+    - [Simpler installation version in dsci100 textbook](https://datasciencebook.ca/move-to-your-own-machine.html)
+    - They can upload a screenshot from the terminal of running our script as proof that it worked.
+
+## 2. Shell
+
+0. Introduction to computing and text-based communication
+    - [521 intro to computing section](https://pages.github.ubc.ca/mds-2021-22/DSCI_521_platforms-dsci_students/materials/lectures/1-lecture1-MDStools-shell-git-github.html#introduction-to-computing)
+1. Use Bash to navigate the computer’s filesystem
+2. Creating new files and directories
+3. Moving, copying, and deleting files and directories
+4. Wildcards
+5. Define and distinguish between absolute file paths and relative file paths
+6. Reading the manual
+    - In general, we can take shell inspiration from [this chapter that I wrote](https://merely-useful.tech/py-rse/bash-basics.html) and [521 Lec 1, bash section](https://pages.github.ubc.ca/mds-2021-22/DSCI_521_platforms-dsci_students/materials/lectures/1-lecture1-MDStools-shell-git-github.html#id1)
+
+## 3. Git and GitHub intro
+
+1. Know of the importance of version control: what it is and why we use it
+2. Use git’s basic functions: clone, add, commit, pull, and push
```

Comment:
Using jupyterlab and the terminal? Do more advanced things in terminal only in the next two chapters.

Reply:
Jlab with git is not explained in 521, so we would need to create material for this.

Reply:
Does jlab +git handle merge conflicts?